src/client/opamArg.ml

```diff
@@ -915,10 +920,17 @@ let build_options =
        actually performing them. \
        WARNING: This option is dangerous and likely to break your opam \
        environment. You probably want `--dry-run'. You've been warned." in
+  let ignore_constraints_on =
+    mk_opt ~section ["ignore-constraints-on"] "PACKAGES"
+      "Forces opam to ignore version constraints on all dependencies to the \
+       listed packages. This can be used to actually test compatibility, but \
+       expect builds to break when using this. Note that version constraints \
+       on optional dependencies and conflicts are unaffected."
```

Comment:
Yes. The only difference would be, with `A` having a depopt on `B >= 2`, that `opam reinstall B --ignore-constraints-on B` would trigger a recompile of `A` even if `B` was `<2`.
I am not sure when or if this may be useful, but probably not in the same cases as ignoring constraints on real depends, which is why I decided not to include depopts.

Of course, if you have scenarios in mind that prove otherwise, I'd be glad to add it.